com.unity.shadergraph/Editor/Data/Nodes/Channel/SwizzleNode.cs

```diff
@@ -27,146 +26,177 @@ public override bool hasPreview
-                blueChannel = TextureChannel.Red;
-            if ((int)alphaChannel >= channelCount)
-                alphaChannel = TextureChannel.Red;
+            AddSlot(new DynamicVectorMaterialSlot(InputSlotId, kInputSlotName, kInputSlotName, SlotType.Input, Vector4.zero));
+            switch (_maskInput.Length)
+            {
+                default:
+                    AddSlot(new Vector4MaterialSlot(OutputSlotId, kOutputSlotName, kOutputSlotName, SlotType.Output, Vector4.zero));
+                    break;
+                case 3:
+                    AddSlot(new Vector3MaterialSlot(OutputSlotId, kOutputSlotName, kOutputSlotName, SlotType.Output, Vector3.zero));
+                    break;
+                case 2:
+                    AddSlot(new Vector2MaterialSlot(OutputSlotId, kOutputSlotName, kOutputSlotName, SlotType.Output, Vector2.zero));
+                    break;
+                case 1:
+                    AddSlot(new Vector1MaterialSlot(OutputSlotId, kOutputSlotName, kOutputSlotName, SlotType.Output, 0));
+                    break;
+            }
+            RemoveSlotsNameNotMatching(new[] { InputSlotId, OutputSlotId });
         }
 
         public void GenerateNodeCode(ShaderStringBuilder sb, GenerationMode generationMode)
         {
-            ValidateChannelCount();
             var outputSlotType = FindOutputSlot<MaterialSlot>(OutputSlotId).concreteValueType.ToShaderString();
             var outputName = GetVariableNameForSlot(OutputSlotId);
             var inputValue = GetSlotValue(InputSlotId, generationMode);
             var inputValueType = FindInputSlot<MaterialSlot>(InputSlotId).concreteValueType;
-            if (inputValueType == ConcreteSlotValueType.Vector1)
-                sb.AppendLine(string.Format("{0} {1} = {2};", outputSlotType, outputName, inputValue));
-            else if (generationMode == GenerationMode.ForReals)
-                sb.AppendLine("{0} {1} = {2}.{3}{4}{5}{6};",
-                    outputSlotType,
-                    outputName,
-                    inputValue,
-                    s_ComponentList[m_RedChannel].ToString(CultureInfo.InvariantCulture),
-                    s_ComponentList[m_GreenChannel].ToString(CultureInfo.InvariantCulture),
-                    s_ComponentList[m_BlueChannel].ToString(CultureInfo.InvariantCulture),
-                    s_ComponentList[m_AlphaChannel].ToString(CultureInfo.InvariantCulture));
+            var InputValueSize = SlotValueHelper.GetChannelCount(inputValueType);
+
+            if (!ValidateMaskInput(InputValueSize))
+            {
+                sb.AppendLine(string.Format("{0} {1} = 0;", outputSlotType, outputName));
+            }
             else
-                sb.AppendLine("{0} {1} = {0}({3}[((int){2} >> 0) & 3], {3}[((int){2} >> 2) & 3], {3}[((int){2} >> 4) & 3], {3}[((int){2} >> 6) & 3]);", outputSlotType, outputName, GetVariableNameForNode(), inputValue);
+            {
```

Comment:
This is great! Much simpler than whatever was happening before, bitshifting is scary 😅